### ISSUE TITLE
No thumpnails on large videos

### ISSUE THREAD

**Issue by kalv78** [hardcore-sushi/DroidFS#204]
Video larger than 200M (not too sure) have no thumpnails 
I tried hidden and non-hidden volumes also on gocryptfs and cryfs but all the same

I did increase the maximum size for thumpnails to 1GB, didn't work

Running DroidFS 2.0.1 on android 11

**Comment by hardcore-sushi** [comment]
Yes, it's because thumbnails are generated with [Glide](https://github.com/bumptech/glide), which doesn't support reading from a custom source. Consequently, the entire file must be decrypted in RAM and given to Glide in order to make a thumbnail. Large files will fill up the memory and the thumbnail generation process will fail. If you look into the logcat, you should see things like:
```
Throwing OutOfMemoryError "Failed to allocate a 106725775 byte allocation with 20828448 free bytes and 19MB until OOM, target footprint 201326592, growth limit 201326592" (VmSize 15140764 kB)
```
I'm still looking at a better way of generating thumbnails. All ideas are welcome.
